sound/soc/sof/pm.c

```diff
@@ -360,6 +361,10 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 			"error: failed to power down DSP during suspend %d\n",
 			ret);
 
+	/* reset ref counts of DSP cores */
+	for (i = 0; i < ARRAY_SIZE(sdev->core_refs); i++)
+		sdev->core_refs[i] = 0;
```

Comment:
Is the mutex held when this function is called?

Reply:
No actually.
Let me add this to make it symmetric with what we do in run_firmware, thanks.

Reply:
can you clarify if this is fixed or not